app/models/topic_tracking_state.rb

```diff
@@ -478,7 +478,10 @@ def self.report_raw_sql(
             NOT (
               #{(skip_new && skip_unread) ? "" : "last_read_post_number IS NULL AND"}
               (
-                COALESCE(category_users.notification_level, #{CategoryUser.default_notification_level}) = #{CategoryUser.notification_levels[:muted]}
+                (
+                  COALESCE(category_users.notification_level, #{CategoryUser.default_notification_level}) = #{CategoryUser.notification_levels[:muted]}
+                  OR topics.category_id IN (#{CategoryUser.indirectly_muted_category_ids_query(user).select("categories.id").to_sql})
```

Comment:
With this change, how does the query plan for the topic tracking state report look like? Since this is a highly performance sensitive query, we need to make sure that this change here does not make the query inefficient.

Reply:
With this new query, do we need the filter above where we check `category_users.notification_level`? It seems like we might be able to simplify the query by just doing `topics.category_id IN (<all categories that are muted by the user>)`.